src/REL/d/a/obj/d_a_obj_tower_hand_D101.cpp

```diff
@@ -1,25 +1,694 @@
 #include "d/a/obj/d_a_obj_tower_hand_D101.h"
 
+#include "common.h"
+#include "d/a/d_a_player.h"
+#include "d/a/obj/d_a_obj_item.h"
+#include "d/col/bg/d_bg_s.h"
+#include "d/col/bg/d_bg_w.h"
+#include "d/flag/sceneflag_manager.h"
+#include "f/f_base.h"
+#include "m/m3d/m_fanm.h"
+#include "m/m_angle.h"
+#include "m/m_mtx.h"
+#include "m/m_vec.h"
+#include "nw4r/g3d/g3d_resfile.h"
+#include "rvl/MTX/mtx.h"
+#include "rvl/MTX/vec.h"
+#include "s/s_Math.h"
+#include "toBeSorted/actor_event.h"
+#include "toBeSorted/event_manager.h"
+#include "toBeSorted/room_manager.h"
+
 SPECIAL_ACTOR_PROFILE(OBJ_TOWER_HAND_D101, dAcOTowerHandD101_c, fProfile::OBJ_TOWER_HAND_D101, 0x180, 0, 6);
 
+const mVec3_c sScales[2] = {mVec3_c(1.0f, 1.0f, 1.0f), mVec3_c(-1.0f, 1.0f, 1.0f)};
+const mVec3_c sBoundingLower = mVec3_c(-1000.0f, 0.0f, -600.0f);
+const mVec3_c sBoundingUpper = mVec3_c(1000.0f, 1000.0f, 1000.0f);
+
+const mVec3_c v1 = mVec3_c::Ex * 50.0f;
+const mVec3_c v2 = mVec3_c::Ey * 50.0f;
+const mVec3_c v3 = mVec3_c::Ez * 50.0f;
+
 STATE_DEFINE(dAcOTowerHandD101_c, RemainOpen);
 STATE_DEFINE(dAcOTowerHandD101_c, Close);
 STATE_DEFINE(dAcOTowerHandD101_c, RemainClosed);
 STATE_DEFINE(dAcOTowerHandD101_c, Open);
 STATE_DEFINE(dAcOTowerHandD101_c, Hold);
 
-void dAcOTowerHandD101_c::initializeState_RemainOpen() {}
-void dAcOTowerHandD101_c::executeState_RemainOpen() {}
+static const char *const sOarcName = "TowerHandD101";
+static const char *const sMdlName = "TowerHandD101";
+static const char *const sAnmName = "TowerHandD101";
+
+static const char *const sCollisionParts[2][3] = {
+    {  "TowerHandD101A",   "TowerHandD101B",   "TowerHandD101C"},
+    {"TowerHandD101A_R", "TowerHandD101B_R", "TowerHandD101C_R"},
+};
+
+static const char *const sLocator1 = "locator1";
+static const char *const sLocator2 = "locator2";
+
+bool dAcOTowerHandD101_c::getScale(int direction, mVec3_c &outScale) {
+    if (!isValidDirectionParam(direction)) {
+        return false;
+    }
+
+    outScale.x = sScales[direction].x;
+    outScale.y = sScales[direction].y;
+    outScale.z = sScales[direction].z;
+    return true;
+}
+
+const mVec3_c &dAcOTowerHandD101_c::getBoundingLower() {
+    return sBoundingLower;
+}
+
+const mVec3_c &dAcOTowerHandD101_c::getBoundingUpper() {
+    return sBoundingUpper;
+}
+
+bool dAcOTowerHandD101_c::isValidDirectionParam(int param) {
+    // TODO
+    bool ok = false;
+    if (param >= 0 && param < 2) {
+        ok = true;
+    }
+
+    return ok;
```

Comment:
Whatever code I write gets turned into branchless comparisons but the original code is not branchless!